pypika/terms.py

```diff
@@ -847,7 +849,7 @@ def fields(self):
         return []
 
     def get_sql(self, **kwargs):
-        dialect = kwargs.get('dialect')
+        dialect = self.dialect or kwargs.get('dialect')
```

Comment:
does it make sense to have two sources for dialect? Where do each of them come from? Is it not redundant?

Reply:
The dialect gets passed to the get_sql() function when the interval is used inside another function. However, just using Interval() by itself doesn't receive the dialect, meaning the default (and incorrect) interval SQL was being returned for the MySQL platform when I was testing.

Reply:
ok, sounds like something we should eventually work out, but maybe not in this PR.